src/NexusMods.DataModel/Abstractions/Entity.cs

```diff
@@ -91,10 +82,13 @@ public void EnsureStored()
     [JsonIgnore]
     public IId DataStoreId
     {
-        get => _id ??= Persist();
+        get => _id ?? throw new UnpersistedEntity();
```

Comment:
Please make a `ThrowHelper` method for this exception, to allow this getter to be inlined.

Throwing exceptions prevents inlining of code, which means that every access to this field suddenly has to go through a method call to a method which is only few instructions at best.